cpp/dolfinx/graph/AdjacencyList.h

```diff
@@ -129,62 +113,47 @@ class AdjacencyList
   /// Equality operator
   bool operator==(const AdjacencyList& list) const
   {
-    return (this->_array == list._array).all()
-           and (this->_offsets == list._offsets).all();
+    return this->_array == list._array and this->_offsets == list._offsets;
   }
 
-  /// Number of nodes
+  /// Gee the number of nodes
   /// @return The number of nodes
-  std::int32_t num_nodes() const { return _offsets.rows() - 1; }
+  std::int32_t num_nodes() const { return _offsets.size() - 1; }
 
   /// Number of connections for given node
   /// @param [in] node Node index
   /// @return The number of outgoing links (edges) from the node
   int num_links(int node) const
   {
-    assert((node + 1) < _offsets.rows());
+    assert((node + 1) < (int)_offsets.size());
     return _offsets[node + 1] - _offsets[node];
   }
 
-  /// Links (edges) for given node
+  /// Get the links (edges) for given node
   /// @param [in] node Node index
   /// @return Array of outgoing links for the node. The length will be
-  ///   AdjacencyList:num_links(node).
-  typename Eigen::Array<T, Eigen::Dynamic, 1>::SegmentReturnType links(int node)
+  /// AdjacencyList:num_links(node).
+  tcb::span<T> links(int node)
```

Comment:
These could have also a `const` return variant?

Reply:
Doesn't really make sense to return by value and use `const`.

Reply:
But `tcb::span<const T> links(int node) const` maybe? (I guess span<T> works like a view that allows writing to the viewed memory if T is not const)

Reply:
Yes, `tcb::span<const T> links(int node) const` has already been added.